test/util/util.bash

```diff
@@ -47,8 +45,46 @@ remove_function() {
     done
 }
 
+# Sudo check function.
 sudo_check() {
     if [[ "${BATS_SKIP_SUDO:+x}" == "x" ]]; then
         skip "skipping, as test requires sudo"
     fi
 }
+
+# APT wrapper for common options that we need.
+apt_path="$(type -pf apt-get)"
+
+apt() {
+    args=()
+    cmd="${1}"
+
+    case "${cmd}" in
+        install) args+=('--allow-downgrades') ;;
+    esac
+
+    apt_cmd=("${apt_path}" "${@}" "${args[@]}")
+
+    if [[ -n "${SUDO_PREFIX}" ]]; then
+        sudo "${apt_cmd[@]}"
+    else
+        "${apt_cmd[@]}"
+    fi
+}
+
+apt-get() {
+    apt "${@}"
+}
+
+# Wrap sudo so we can capture it's APT calls.
+sudo() {
+    cmd="${1}"
+
+    case "${cmd}" in
+        apt|apt-get) SUDO_PREFIX=y apt "${@:2}" ;;
+    esac
+}
+
+export -f apt apt-get sudo
+
+# vim: set ts=4 sw=4 expandtab:
```

Comment:
This makes it so that any calls to `apt install` or `apt-get install` automatically get ran with `--allow-downgrades`, as the CI was failing on some lowered package versions in the CI previously (test packages are recreated throughout the tests and then installed, and sometimes the pkgname's for the packages are the same, which caused the lowered version issue).